tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -5953,3 +5981,858 @@ def get_prod(arr):
+
+    cmplx128_from_float32 = uniform(low=min_float32 / 10, high=max_float32 / 10, size=size) + uniform(low=min_float32 / 10, high=max_float32 / 10, size=size) * 1j
+    # the result of the last operation is a Python complex type which has 8 bytes in the alignment,
+    # that's why we need to convert it to a numpy.complex64 the needed type.
+    cmplx64 = np.complex64(cmplx128_from_float32)
+    cmplx128 = uniform(low=min_float64 / 10, high=max_float64 / 10, size=size) + uniform(low=min_float64 / 10, high=max_float64 / 10, size=size) * 1j
+
+    f_bl = epyccel(get_linspace, language=language)
+
+    assert (f_bl(bl) == get_linspace(bl))
+
+    f_integer8 = epyccel(get_linspace, language=language)
+    f_integer16 = epyccel(get_linspace, language=language)
+    f_integer = epyccel(get_linspace, language=language)
+    f_integer32 = epyccel(get_linspace, language=language)
+
+    assert (f_integer8(integer8) == get_linspace(integer8))
+    assert (f_integer16(integer16) == get_linspace(integer16))
+    assert (f_integer(integer) == get_linspace(integer))
+    assert (f_integer32(integer32) == get_linspace(integer32))
+
+    f_integer64 = epyccel(get_linspace, language=language)
+    assert (f_integer64(integer64) == get_linspace(integer64))
+
+    f_fl = epyccel(get_linspace, language=language)
+    f_fl32 = epyccel(get_linspace, language=language)
+    f_fl64 = epyccel(get_linspace, language=language)
+
+    assert (f_fl(fl) == get_linspace(fl))
+    assert (f_fl32(fl32) == get_linspace(fl32))
+    assert (f_fl64(fl64) == get_linspace(fl64))
+
+    f_complex64 = epyccel(get_linspace, language=language)
+    f_complex128 = epyccel(get_linspace, language=language)
+
+    assert (f_complex64(cmplx64) == get_linspace(cmplx64))
+    assert (f_complex128(cmplx128) == get_linspace(cmplx128))
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [pytest.mark.fortran,
+            pytest.mark.skip(reason="Still under maintenance, See #771")]),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Needs a C printer see https://github.com/pyccel/pyccel/issues/791"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = [
+            pytest.mark.python]
+        )
+    )
+)
+
+
+def test_numpy_linspace_array_like_2d(language):
+
+    @types('bool[:,:]')
+    @types('int[:,:]')
+    @types('int8[:,:]')
+    @types('int16[:,:]')
+    @types('int32[:,:]')
+    @types('int64[:,:]')
+    @types('float[:,:]')
+    @types('float32[:,:]')
+    @types('float64[:,:]')
+    @types('complex64[:,:]')
+    @types('complex128[:,:]')
+    def get_linspace(arr):
+        from numpy import linspace, shape, ones
+        numberOfSamplesToGenerate = 7
+        start = ones((2,5))
+        stop = arr
+        a = linspace(start, stop, numberOfSamplesToGenerate)
+        s = shape(a)
+        return len(s), s[0], s[1], s[2], a[0, 0, 0], a[0, 1, 0], a[1, 0, 0], a[1, 1, 0]
```

Comment:
Do you think this output is sufficient to test that linspace does everything it should? Would it be possible to return the min and max to help double check? It would also be good to check that `a` is ordered correctly if possible